app/models/event.rb

```diff
@@ -147,6 +147,41 @@ def ready_to_schedule?(schedule_action)
     return true
   end
 
+  def self.staffing_level
+    begin
+      relevant_events = self.current_or_next_events
+      event_staffing = relevant_events.map { |event| event.staffing_level }
+      event_staffing.min
+    rescue => e
+      puts e
+      -1
+    end
+  end
+
+  def self.current_or_next_events
+    current_events = self.current_events
+    events = current_events.empty? ? [self.next_event] : current_events
+  end
+
+  def self.current_events
+    self.where("start_time < ?", Time.now)
+        .where("end_time > ?", Time.now)
+  end
+
+  def self.next_event
+    next_event = self.where("start_time > ?", Time.now)
+        .order("start_time ASC")
+        .first
+    raise '---Error: There are no events---' unless next_event
+    next_event
+  end
+
+  def staffing_level
+    tasks = self.tasks
+    raise '---Error: Event has no tasks---' if tasks.empty?
```

Comment:
Conceptually, not sure this is an error. It seems reasonable that a new event would have no tasks.

Reply:
Agreed.  Though it is not clear what the staffing_level of an event with no tasks should be.  is it full or empty?